bin/run-tests.sh

```diff
@@ -31,12 +31,12 @@ for test_dir in "${tmp_dir}"/*; do
 
     bin/run.sh "${test_dir_name}" "${test_dir_path}" "${test_dir_path}"
 
-    for file in "$results_file_path" "$expected_results_file_path"; do
-        # We sort both the '.message' values in results.json and expected_results.json files
-        tmp_file=$(mktemp -p "$test_dir/")
-        sorted_message=$(cat $file | jq -r '.message' >"$tmp_file" && sort "$tmp_file")
-        jq --arg msg "$sorted_message" '.message = $msg' "$file" >"$tmp_file" && mv "$tmp_file" "$file"
-    done
+    has_message=$(jq 'has("message") and .message != null' "$results_file_path")
+
+    if [ "$has_message" = "true" ]; then
+        jq '.message = (.message|split("\n")|map(sub("^[ \t]+"; "")|select(. != ""))|sort|join("\n"))' "$results_file_path" > "$results_file_path.tmp"
+        mv "$results_file_path.tmp" "$results_file_path"
+    fi
```

Comment:
```suggestion
    jq '.message = (
          .message // "" | split("\n") | map(ltrim | select(. != "")
        ) | sort | join("\n")) |
        if .message == "" then del(.message) end
    ' "$results_file_path" > "$results_file_path.tmp"
    mv "$results_file_path.tmp" "$results_file_path"
```

This works, too :) No need for multiple `jq` commands of shell conditions.

Note, this requires `jq` 1.7+ for the `if` line. On older `jq` versions, you need an `else . end`